src/Compilers/Core/MSBuildTask/BuildClient.cs

```diff
@@ -300,6 +303,35 @@ private static int HandleResponse(BuildResponse response)
         private const int MAX_PATH_SIZE = 260;
 
         /// <summary>
+        /// Try all processes that start with the current pipe name and return
+        /// the connected pipe if one is found. Otherwise, return null.
+        /// <paramref name="newPipeName" /> will contain the next free pipe
+        /// name if no connection was made.
+        /// </summary>
+        private static NamedPipeClientStream TryAllProcesses(
+            string pipeName,
+            int timeoutMs,
+            CancellationToken cancellationToken,
+            out string newPipeName)
+        {
+            string basePipeName = pipeName;
+            for (int counter = 1; File.Exists(pipeName); counter++)
+            {
+                NamedPipeClientStream pipe;
+                if (null != (pipe = TryConnectToProcess(pipeName, timeoutMs, cancellationToken)))
+                {
+                    newPipeName = pipeName;
+                    return pipe;
+                }
+
+                // Append an integer counter to the pipename
+                pipeName = basePipeName + counter.ToString(CultureInfo.InvariantCulture);
```

Comment:
Perhaps there should be a separator char (period, probably) so that it's obviously a suffix. i.e.
```pipeName = basePipeName + "." + counter.ToString(etc.```